cmd/agent/subcommands/run/command.go

```diff
@@ -341,8 +341,7 @@ func getSharedFxOption() fx.Option {
 		fx.Supply(dogstatsdServer.Params{
 			Serverless: false,
 		}),
-		forwarder.Bundle(),
-		fx.Provide(func(config config.Component, log log.Component) defaultforwarder.Params {
+		forwarder.BundleWithProvider(func(config config.Component, log log.Component) defaultforwarder.Params {
```

Comment:
I think adding the bundle with proider deviates from the original intentend that is to provide an easy more straight forward to pass parameters to component module or bundle functions. 

I think for this case it seems that the forwarder does not fit nicely into that idea. But having a dedicated function that takes to two arguments, means that if in the future we need to add a third argument we need to update the function signature. 

Thinking out loud. 

Could we do something like:

```go
		fx.Provide(func(config config.Component, log log.Component) defaultforwarder.Params {
			params := defaultforwarder.NewParams(config, log)
			// Enable core agent specific features like persistence-to-disk
			params.Options.EnabledFeatures = defaultforwarder.SetFeature(params.Options.EnabledFeatures, defaultforwarder.CoreFeatures)
			return params
		}),
		fx.Provide(func(params defaultforwarder.Params) defaultforwarder.Component {
			return defaultforwarder.Module(params)
		}),
```

Reply:
> I think adding the bundle with proider deviates from the original intentend that is to provide an easy more straight forward to pass parameters to component module or bundle functions.

This is a very good question. I implemented this solution with two main ideas in mind:
1. Make it obvious that defaultforwarder needs a parameter
2. Remove `fx` from `command.go` to make the code more obvious to people not familiar with fx.

> means that if in the future we need to add a third argument we need to update the function signature.

If we need to add a third argument, we have to update the code at several locations. I don't think it is an issue to also update the signature.